src/services/coreBackService.ts

```diff
@@ -0,0 +1,115 @@
+import { createPersonalDataHeaders } from "@govuk-one-login/frontend-passthrough-headers";
+import { createAxiosInstance } from "../app/shared/axiosHelper";
+import config from "../lib/config";
+import { Request } from "express";
+import { AxiosRequestConfig, AxiosResponse } from "axios";
+import { Logger } from "pino";
+
+const axiosInstance = createAxiosInstance(config.API_BASE_URL);
+
+// Extend axios definition with logger
+declare module "axios" {
+  interface AxiosRequestConfig {
+    logger?: Logger;
+  }
+}
+
+export interface InitialiseSessionRequest {
+  responseType: string;
+  clientId: string;
+  redirectUri: string;
+  state: string;
+  scope: string;
+  request: string;
+}
+
+export interface CriCallbackRequest {
+  authorizationCode?: string;
+  credentialIssuerId: string;
+  error?: string;
+  errorDescription?: string;
+  redirectUri: string;
+  state?: string;
+}
+
+function generateAxiosConfig(url: string, req: Request): AxiosRequestConfig {
```

Comment:
another q for me - there's a mix of regular `function` and arrow functions in this repo, is there a preference going forward?

Reply:
This is another question of taste, and I don't think there's a perfect answer... The semantics are slightly different.

For expressions (e.g. `arr.map((a) => b)` vs. `arr.map(function(a) { return b; })`, I would always use an arrow function, unless I need `this` binding (which is fairly rare these days).

For top-level definitions, it's a bit more debatable - I prefer using const for a couple of reasons:
- Consistency - that way _all_ your functions are arrow functions (unless for a good reason, which signals that there's something funky about the function)
- It's more functional - that way your 'functions' are just variables like everything else

The last point is particularly relevant with Typescript definitions where you want to use a function type:
```
// function declaration - needs to use individual param/return types
function someRequestHandler(req: Request, res: Response, next: NextFunction): Response {
  ...
}

// arrow function variable - typed as a handler
const someRequestHandler: RequestHandler = (req, res, next) => {
  ...
}

// or the awkward middle-ground - which is kinda the worst of both worlds
const someRequestHandler: RequestHandler = function (req, res, next) {
  ...
}
```

I can change this for consistency, and we could consider adding an eslint rule to enforce it!

Reply:
I didn't know you couldn't use function types with `function`, thanks for the insight :)